cunumeric/config.py

```diff
@@ -68,15 +237,15 @@ def get_resource_configuration(self):
         config.max_shardings = 0
         return config
 
-    def destroy(self):
+    def destroy(self) -> None:
         if self.runtime is not None:
             self.runtime.destroy()
 
 
 CUNUMERIC_LIB_NAME = "cunumeric"
 cunumeric_lib = CuNumericLib(CUNUMERIC_LIB_NAME)
 cunumeric_context = get_legate_runtime().register_library(cunumeric_lib)
-_cunumeric = cunumeric_lib.shared_object
+_cunumeric = cast(_CunumericSharedLib, cunumeric_lib.shared_object)
```

Comment:
I think some re-arrangement of the initialization code would be required to try to make it clear that `shared_object` is not None at this point.